### ISSUE TITLE
Spreadsheet: configuration table doesn't handle сonditionals expressions correctly

### ISSUE THREAD

**Issue by kaiwas** [FreeCAD/FreeCAD#14581]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

I'm trying to create an object based on interchangeable sketches.
All data will be filled into a spreadsheet.
Considering that there can be a lot of possible types of sections, I tried to make a filter using сonditionals expressions.
At first glance, everything works as it should: the sections change and the list is correct.
But despite this, data is taken only from the top two rows (because the final list has only two options).
That is, the regular expression displays the headers correctly, but does not take the data itself from the cells I need.

I show the algorithm in the video.

https://github.com/FreeCAD/FreeCAD/assets/32273833/00578854-5601-4b3c-99f1-b9ecc6388cae

Test file is attached. Rename TXT --> FCstd
[SuperObject_ENG.txt](https://github.com/user-attachments/files/15759759/SuperObject_ENG.txt)

сonditional expression: `TIP[0] == 0 ? Spreadsheet.cells[<<A4:A5>>] : Spreadsheet.cells[<<A6:A7>>]`

The main goal is to hide all unnecessary sections, because in the end there can be a lot of them and the list “without a filter” can be very large and inconvenient.
If the filter works correctly, you will be able to select only what you need, and all unnecessary data will be hidden from the selection.

### Full version info

```shell
0.22
```

### Subproject(s) affected?

Expressions

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Roy-043** [comment]
That is quite creative. :thumbsup:

This workflow was probably never considered. The enumeration value of the `Number` property (displayed in `INFO`) is applied to the whole spreadsheet cell range, and not to the range specified by the conditional expression.

As an alternative you can remove the TIP property and display a submenu for the `Number` property by adding 'group' names in the spreadsheet:
```
Quade | Quade 50х60
Quade | Quade 80х30
Triangle | Triangle 150х180х250
Triangle | Triangle  300х400х500
```
See:
https://wiki.freecad.org/FeaturePython_Custom_Properties#App::PropertyEnumeration

![enum-submenu](https://github.com/FreeCAD/FreeCAD/assets/70520633/119bf73a-2828-4641-8de1-34368b1fda70)

**Comment by kaiwas** [comment]
> As an alternative you can remove the `TIP` property and display a submenu for the `Number` property by adding 'group' names in the spreadsheet:

Great! Thank you!
This will be the way out.
Otherwise I was already desperate to do what I had in mind. You saved me. I'll definitely try it.

In my execution, the conditional expression outputs 0,1,2,3 and not 3,4,5,6
I understand that this may not be an error, but due to the fact that the list is displayed correctly, the selection corresponding to the list was also expected.

**Comment by kaiwas** [comment]
> remove the `TIP` property

Although I will have to think a little about implementing my idea.
The main goal was that with the help of conditional expressions it would be possible to “hide” everything that does not belong to a specific profile. In fact, after selecting a profile, only one group should remain.

But perhaps I need to look at the problem from the other side and write the expression in the “TIP” property. (depending on the selected group)

**Comment by kaiwas** [comment]
It gave me my strength back!
Works great. Perhaps even better than I previously thought)))
I am attaching a file for testing. 
![Peek 2024-06-08 00-28](https://github.com/FreeCAD/FreeCAD/assets/32273833/334760d4-bda4-4bc0-a8b0-54dede08d6a8)
[SuperObject_7_new_logic.txt](https://github.com/user-attachments/files/15828107/SuperObject_7_new_logic.txt)
Rename TXT --> FCstd

As soon as I bring the idea to completion, I will post it on the forum for open access.

**Comment by FEA-eng** [comment]
> It gave me my strength back! Works great. Perhaps even better than I previously thought)))

Can the issue be closed then?

**Comment by kaiwas** [comment]
> Can the issue be closed then?

No. Roy043 suggested a workaround, but it doesn't solve the original problem with expressions.

I'll check later to see if anything has changed in the program. I'll post it here.

**Comment by kaiwas** [comment]
The situation hasn't changed. Expressions work as before.

**Comment by FEA-eng** [comment]
It should still be relevant
